File: src/main/java/com/randomperson22/piggymodtest/npc/pony/PonyNpcModel.java
package com.randomperson22.piggymodtest.npc.pony;

import net.minecraft.util.ResourceLocation;
import software.bernie.geckolib3.model.AnimatedGeoModel;

public class PonyNpcModel extends AnimatedGeoModel<EntityPonyNpc> {
	
    public ResourceLocation getModelLocation(EntityPonyNpc entity) {
        return new ResourceLocation("piggymodtest", "geo/ponynpc.geo.json");
    }

    public ResourceLocation getTextureLocation(EntityPonyNpc entity) {
        return new ResourceLocation("piggymodtest", "textures/entities/ponynpc.png");
    }

    public ResourceLocation getAnimationFileLocation(EntityPonyNpc entity) {
        return new ResourceLocation("piggymodtest", "animations/doggynpc.animation.json");
    }
}

File: src/main/kotlin/com/randomperson22/piggymodtest/npc/pony/PonyNpcModel.kt
package com.randomperson22.piggymodtest.npc.pony

import net.minecraft.util.ResourceLocation
import software.bernie.geckolib3.model.AnimatedGeoModel

class PonyNpcModel : AnimatedGeoModel<EntityPonyNpc?>() {
    override fun getModelLocation(entity: EntityPonyNpc?): ResourceLocation {
        return ResourceLocation("piggymodtest", "geo/ponynpc.geo.json")
    }

    override fun getTextureLocation(entity: EntityPonyNpc?): ResourceLocation {
        return ResourceLocation("piggymodtest", "textures/entities/ponynpc.png")
    }

    override fun getAnimationFileLocation(entity: EntityPonyNpc?): ResourceLocation {
        return ResourceLocation("piggymodtest", "animations/doggynpc.animation.json")
    }
}